crates/polars-ops/src/chunked_array/interpolate.rs

```diff
@@ -52,10 +52,10 @@ fn signed_interp<T>(low: T, high: T, steps: IdxSize, steps_n: T, av: &mut Vec<T>
 where
     T: Sub<Output = T> + Mul<Output = T> + Add<Output = T> + Div<Output = T> + NumCast + Copy,
 {
-    let diff = high - low;
+    let slope = (high - low) / steps_n;
     for step_i in 1..steps {
         let step_i: T = NumCast::from(step_i).unwrap();
-        let v = linear_itp(low, step_i, diff, steps_n);
+        let v = linear_itp(low, step_i, slope);
```

Comment:
`diff / steps_n` is independent of the loop, it doesn't need recomputing each time

Reply:
Good observation :+1: